### PR TITLE
chore(edit-content): Thumbnail in the Binary Preview Punchlist #26907

### LINKED ISSUE #26907: Edit Content: Thumbnail in the Binary Preview Punchlist
### Parent
#25445 

### User Story
As a content creator, I want to be able to render PDF thumbnails in the binary preview, so I can easily identify PDF files.

```[tasklist]
### Tasks
- [x] Use the `/dA/<%=inode%>/<%=field.getVelocityVarName()%>/500mw/20q/<%=fileName%>` URL to render ALL image thumbnails
- [x] The thumbnail should be properly sized and rendered
- [x] The thumbnail should accurately reflect the content of the PDF
- [x] Label to show image width and height as W x H, not W,H
- [x] font size in monaco editor should  compute to 14px, not 12px
- [x] File size (Bytes) should be converted to kb/mb/gb based on the file size (meaning should be human friendly). Also bytes are not working locally for me. Is anyone else seeing this issue? (we can do a pipe)
- [x] To maintain consistency, use the `dot-content let-thumbnail` component in the Binary field
- [x] Enable editing when the user clicks on editable files.
- [x] Close Monaco Editor on `ESC` without closing the entire dialog
- [x] Enable save on Binary Field on the old Edit Content
```

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
N/A

### Assumptions & Initiation Needs
- The PDF file must be properly formatted
- Appropriate memory resources should be available

### Quality Assurance Notes & Workarounds
- Thumbnail should be properly rendered on all major browsers

### Sub-Tasks & Estimates
N/A

### PR DESCRIPTION
**Pull request by rjvelazco**
## Changes

1. Use the correct thumbnail path (`/dA/<%=inode%>/500mw/20q/<%=fileName%>`)
2. Add pdf thumbnail.
3. Implement new attribute `editableAsText`.
4. Update font size in Monaco editor from 12px to 14px.
5. Show File Size in kb/mb/gb to be human-friendly.

### Image

https://github.com/dotCMS/core/assets/72418962/e7603c2d-dbfa-4aae-9aa3-31b0c41c8f80

image-path

### PDF

https://github.com/dotCMS/core/assets/72418962/75925f9d-b141-44a9-83d4-2f5e4fcc7d93

### Video

https://github.com/dotCMS/core/assets/72418962/5d2363e9-5815-4753-9b77-19f4bcf443a4

### Editable File

https://github.com/dotCMS/core/assets/72418962/42828443-2029-42f4-9932-0528bdb5739f

## Edit Image (Legacy)

https://github.com/dotCMS/core/assets/72418962/a7fc54ad-f8d4-4810-8bcb-1ffc34ebc70f

### Editing an image multiple times
https://github.com/dotCMS/core/assets/72418962/8a84ac4e-6404-4f82-a453-63cb4a6b28d3

### CODE REVIEW

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/dot-edit-content-binary-field.component.ts [hunk 1]
```diff
@@ -152,15 +154,18 @@ export class DotEditContentBinaryFieldComponent
                 tap(() => this.dotBinaryFieldStore.setStatus(BinaryFieldStatus.UPLOADING)),
                 delay(500) // Loading animation
```
Comment (zJaaal) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/dot-edit-content-binary-field.component.ts:155]: do we want this delay?

File: core-web/libs/ui/src/lib/components/dot-content-thumbnail/dot-content-thumbnail.component.ts [hunk 2]
```diff
@@ -18,12 +18,12 @@ export enum CONTENT_THUMBNAIL_TYPE {
...
+    name: string;
+    contentType: string;
     tempUrl?: string;
-    inode?: string;
-    name?: string;
-    contentType?: string;
-    iconSize?: string;
-    titleImage?: string;
+    iconSize?: string; // Remove
+    titleImage?: string; //
```
Comment (zJaaal) [inline core-web/libs/ui/src/lib/components/dot-content-thumbnail/dot-content-thumbnail.component.ts:26]: Do we need this comments?

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/dot-edit-content-binary-field.component.ts (hunk 1 above)
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/dot-edit-content-binary-field.component.ts:155]: The last time I talked to @fmontes we agreed on that. Do we want to keep it? @fmontes

File: core-web/libs/ui/src/lib/components/dot-content-thumbnail/dot-content-thumbnail.component.ts (hunk 2 above)
Comment (rjvelazco) [reply core-web/libs/ui/src/lib/components/dot-content-thumbnail/dot-content-thumbnail.component.ts:26]: I'll remove it

**Review by zJaaal — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**